### PR TITLE
API 4.0: Telegram Passport - WIP

### PR DESCRIPTION
**Pull request by tuscen**
This PR adds Telegram Passport support.

#758

### CODE REVIEW

File: src/Telegram.Bot/Types/Enums/MessageType.cs [hunk 1]
```diff
@@ -142,6 +143,12 @@ public enum MessageType
...
         /// The <see cref="Message"/> contains non-default <see cref="Message.MigrateToChatId"/>
         /// </summary>
-        MigratedFromGroup
+        MigratedFromGroup,
+
+        /// <summary>
+        /// The <see cref="Message"/> contains non-default <see cref="Message.PassportData"/>
+        /// </summary>
+        [EnumMember(Value = "passport_data")]
+        PassportData
```
Comment (pouladpld) [inline src/Telegram.Bot/Types/Enums/MessageType.cs]: We are trying not to use enum values anymore.

File: src/Telegram.Bot/Types/Message.cs [hunk 2]
```diff
@@ -382,6 +389,9 @@ public MessageType Type
                 if (MigrateToChatId != default)
                     return MessageType.MigratedToSupergroup;
 
+                if (PassportData != default)
```
Comment (pouladpld) [inline src/Telegram.Bot/Types/Message.cs]: We are trying not to use enum values anymore.

**Comment by pouladpld** [comment]
This PR won't be merged because scope is pretty big and changes min supported netstandard. I'll cherrypick some of the commits at #778

**Comment by MihaZupan** [comment]
Some of the features here are only available in the higher target. Min target stays the same

File: src/Telegram.Bot/Enums/Gender.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        /// <summary>
+        /// Gender, male or female
+        /// </summary>
+        public static class Gender
+        {
+            /// <summary>
+            /// Male
+            /// </summary>
+            public const string male = nameof(male);
```
Comment (tuscen) [inline src/Telegram.Bot/Enums/Gender.cs]: Are you sure about that? Public static fields should be in pascal case according to the MS [style guide](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-fields).

Comment (pouladpld) [reply src/Telegram.Bot/Enums/Gender.cs]: I was looking for readability but OK, I change it.

**Comment by pouladpld** [comment]
#788

**Pull request closed without merging**